tests/trace/test_client_trace.py

```diff
@@ -656,6 +657,57 @@ def test_trace_call_query_filter_trace_roots_only(client):
         assert len(inner_res.calls) == exp_count
 
 
+def test_trace_call_query_filter_wb_run_ids(client):
+    run_name = "test-run-name"
+    full_wb_run_id = f"{client.entity}/{client.project}/{run_name}"
+    from weave.trace import weave_client
+
+    with mock.patch.object(
+        weave_client, "safe_current_wb_run_id", lambda: full_wb_run_id
+    ):
+        call_spec_1 = simple_line_call_bootstrap()
+    call_spec_2 = simple_line_call_bootstrap()
+
+    for wb_run_ids, exp_count in [
+        (None, call_spec_1.total_calls + call_spec_2.total_calls),
+        ([], call_spec_1.total_calls + call_spec_2.total_calls),
+        ([full_wb_run_id], call_spec_1.total_calls),
+        ([f"{client.entity}/{client.project}/NOT_A_CALL"], 0),
+    ]:
+        inner_res = get_client_trace_server(client).calls_query(
+            tsi.CallsQueryReq(
+                project_id=get_client_project_id(client),
+                filter=tsi.CallsFilter(wb_run_ids=wb_run_ids),
+            )
+        )
+
+        assert len(inner_res.calls) == exp_count
+
+
+def test_trace_call_query_filter_wb_user_ids(client):
+    call_spec_1 = simple_line_call_bootstrap()
+
+    # OMG! How ugly is this?! The layers of testing servers is nasty
+    client.server.server._next_trace_server._user_id = "second_user"
+
+    call_spec_2 = simple_line_call_bootstrap()
+
+    for wb_user_ids, exp_count in [
+        (None, call_spec_1.total_calls + call_spec_2.total_calls),
+        ([], call_spec_1.total_calls + call_spec_2.total_calls),
+        (["second_user"], call_spec_2.total_calls),
+        (["NOT_A_USER"], 0),
```

Comment:
Might be nice to test the case where more than one Run ID / User ID are specified as part of the filter.

Reply:
added